doc/sphinx/proofs/automatic-tactics/logic.rst

```diff
@@ -130,47 +141,41 @@ some incompatibilities.
 
       Prints the default tactic used by :tacn:`firstorder` when no rule applies.
 
-.. tacv:: firstorder @tactic
-
-   Tries to solve the goal with :n:`@tactic` when no logical rule may apply.
-
 .. tacv:: firstorder using {+ @qualid}
 
    .. deprecated:: 8.3
 
-      Use the syntax below instead (with commas).
-
-.. tacv:: firstorder using {+, @qualid}
-
-  Adds lemmas :n:`{+, @qualid}` to the proof-search environment. If :n:`@qualid`
-  refers to an inductive type, it is the collection of its constructors which are
-  added to the proof-search environment.
+      .. todo PR Can we drop the old syntax now?
 
-.. tacv:: firstorder with {+ @ident}
-
-  Adds lemmas from :tacn:`auto` hint bases :n:`{+ @ident}` to the proof-search
-  environment.
+      Use the syntax below instead (with commas).
 
-.. tacv:: firstorder @tactic using {+, @qualid} with {+ @ident}
+.. opt:: Firstorder Depth @natural
 
-  This combines the effects of the different variants of :tacn:`firstorder`.
+   This option controls the proof search depth bound.
 
-.. opt:: Firstorder Depth @natural
-   :name: Firstorder Depth
+.. tacn:: congruence {? @natural } {? with {+ @one_term } }
 
-   This option controls the proof-search depth bound.
+   :token:`natural`
```

Comment:
It's important that the description of the optional clauses comes *after* the two paragraphs describing the tactic. Especially, the first clause naturally follows the explanation from the second paragraph.

Reply:
I kept the first two sentences where they were and moved the rest.  Take a look.

Reply:
I don't see anything changed (even when looking at the review commit).

Reply:
I remember making the change, not sure how it was lost.